CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyWorker.cc

```diff
@@ -735,20 +778,21 @@ void SiStripHitEfficiencyWorker::fillForTraj(const TrajectoryAtInvalidHit& tm,
           << ", withinAcceptance=" << withinAcceptance;
 
       unsigned int layer = TKlayers;
-      if (showRings_ && layer > 10) {        // use rings instead of wheels
-        if (layer < 14) {                    // TID
-          layer = 10 + ((iidd >> 9) & 0x3);  // 3 disks and also 3 rings -> use the same container
-        } else {                             // TEC
-          layer = 13 + ((iidd >> 5) & 0x7);
+      if (showRings_ && layer > bounds::k_LayersAtTOBEnd) {  // use rings instead of wheels
+        if (layer <= bounds::k_LayersAtTIDEnd) {             // TID
+          layer = bounds::k_LayersAtTOBEnd +
+                  tTopo->tidRing(iidd);  // ((iidd >> 9) & 0x3);  // 3 disks and also 3 rings -> use the same container
+        } else {                         // TEC
+          layer = bounds::k_LayersAtTIDEnd + tTopo->tecRing(iidd);  // ((iidd >> 5) & 0x7);
         }
       }
       unsigned int layerWithSide = layer;
-      if (layer > 10 && layer < 14) {
-        const auto side = (iidd >> 13) & 0x3;  // TID
+      if (layer > bounds::k_LayersAtTOBEnd && layer <= bounds::k_LayersAtTIDEnd) {
+        const auto side = tTopo->tidSide(iidd);  //(iidd >> 13) & 0x3;  // TID
         if (side == 2)
           layerWithSide = layer + 3;
-      } else if (layer > 13) {
-        const auto side = (iidd >> 18) & 0x3;  // TEC
+      } else if (layer > bounds::k_LayersAtTIDEnd) {
+        const auto side = tTopo->tecSide(iidd);  // (iidd >> 18) & 0x3;  // TEC
```

Comment:
in case if you change things for `k_LayersAtTECEnd` earlier, would you mind cleaning the commented out code here?